### ISSUE TITLE
README example bug: Converting circular structure to JSON

### ISSUE THREAD

**Issue by oMaten** [GoogleChrome/lighthouse#1744]
**version:** 1.5.1
**node:** 6.9.2

I set flags like `{ output: 'json' }` and run programmatically example.

when printer write the file,  error happened as follow
`(node:91954) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Converting circular structure to JSON`

**Comment by oMaten** [comment]
By the way, in the programmatically example.
```js
function launchChromeAndRunLighthouse(url, flags, config) {
  const launcher = new ChromeLauncher({ port: 9222, autoSelectChrome: true });

  launcher.isDebuggerReady() // return launcher.isDebuggerReady()
    .catch(() => {
      if (flags.skipAutolaunch) {
        return;
      }
      return launcher.run(); // Launch Chrome.
    }

  //...

```

should return `launcher`, otherwise `launchChromeAndRunLighthouse` is not a promise

**Comment by ebidel** [comment]
@oMaten, fixed that readme bug.

**Comment by ssunny2** [comment]
Hi,
I started getting the same issue when I integrated lighthouse with webpack. It initially worked fine and was generating a report in Chrome, but now after serving, it fails after printing the result on node console.

**webpack.dev.js**
const WebpackLighthousePlugin = require('webpack-lighthouse-plugin');

plugins: [
       new WebpackLighthousePlugin({
         url: 'https://www.google.com'
       }),
...
]

**error trace in console**
..
..
_Fri, 22 Dec 2017 05:38:41 GMT status Evaluating: Links have descriptive text.
Fri, 22 Dec 2017 05:38:41 GMT status Evaluating: Page isn’t blocked from indexing
Fri, 22 Dec 2017 05:38:41 GMT status Evaluating: Document has a valid `hreflang`
Fri, 22 Dec 2017 05:38:41 GMT status Generating results...
Runtime error encountered: TypeError: Converting circular structure to JSON
    at Object.stringify (native)
    at ReportGeneratorV2.generateReportHtml (C:\Users\sherin.sunny\git\ng-ui\node_modules\lighthouse\lighthouse-core\report\v2\report-generator.js:120:32)
    at createOutput (C:\Users\sherin.sunny\git\ng-ui\node_modules\lighthouse\lighthouse-cli\printer.js:46:34)
    at Promise (C:\Users\sherin.sunny\git\ng-ui\node_modules\lighthouse\lighthouse-cli\printer.js:100:20)
    at Object.write (C:\Users\sherin.sunny\git\ng-ui\node_modules\lighthouse\lighthouse-cli\printer.js:98:10)
    at promise.then._ (C:\Users\sherin.sunny\git\ng-ui\node_modules\webpack-lighthouse-plugin\src\lighthouse-bin.js:58:24)
    at process._tickCallback (internal/process/next_tick.js:109:7)
TypeError: Converting circular structure to JSON
    at Object.stringify (native)
    at ReportGeneratorV2.generateReportHtml (C:\Users\sherin.sunny\git\ng-ui\node_modules\lighthouse\lighthouse-core\report\v2\report-generator.js:120:32)
    at createOutput (C:\Users\sherin.sunny\git\ng-ui\node_modules\lighthouse\lighthouse-cli\printer.js:46:34)
    at Promise (C:\Users\sherin.sunny\git\ng-ui\node_modules\lighthouse\lighthouse-cli\printer.js:100:20)
    at Object.write (C:\Users\sherin.sunny\git\ng-ui\node_modules\lighthouse\lighthouse-cli\printer.js:98:10)
    at promise.then._ (C:\Users\sherin.sunny\git\ng-ui\node_modules\webpack-lighthouse-plugin\src\lighthouse-bin.js:58:24)
    at process._tickCallback (internal/process/next_tick.js:109:7)_

Lighthouse recently added another Category called 'SEO' along with the original 4 checks(PWA,Performance, Accessibility, Best Practices). Can anybody confirm if its because of this addition that I'm getting an error in generating reports?

**Comment by patrickhulce** [comment]
Thanks @ssunny2 this is actually because webpack lighthouse doesn't remove the artifacts from the results before creating the report which it should have been doing. Filed a PR to fix this https://github.com/addyosmani/webpack-lighthouse-plugin/pull/10

**Comment by ev1stensberg** [comment]
@ssunny2 Fixed in [`v 1.0.4`](https://github.com/addyosmani/webpack-lighthouse-plugin/commit/9ced38cade7240390840e811b3acc13a4980b6b1), and it should be on npm shortly!
